### ISSUE TITLE
Assigning `{}` should zero struct padding bytes as well

### ISSUE THREAD

**Issue by Tetralux** [odin-lang/Odin#789]
* Operating System: Windows and Linux x64
* Please paste `odin version` output: `odin version 0.13.1-a55568b0`

Default-initializing a struct with `{}` leaves padding bytes untouched, but it should zero them as well.

On Linux:
- With `-opt:1` (or greater).

On Windows (_on old backend_):
- With `-opt:1` (or greater).

```odin
package test

import "core:mem"
import "core:fmt"

main :: proc() {
    using fmt;

    S :: struct {
        i: i32,
        u: u64,
    };

    pb :: proc(s: ^S) {
        data := mem.ptr_to_bytes(s, 1);
        fmt.printf("%x\n", data);
    }

    s: S;
    pb(&s); println(s); println();

    data := mem.ptr_to_bytes(&s, 1);
    for b in &data {
        b = 0xaa;
    }
    pb(&s); println(s); println();

    s = {};
    pb(&s); println(s); println();
    /*
    [0, 0, 0, 0, aa, aa, aa, aa, 0, 0, 0, 0, 0, 0, 0, 0]
    S{i = 0, u = 0}
    */
}
```

**Comment by gingerBill** [comment]
Is this still happening?

**Comment by gingerBill** [comment]
Could you please test this on the latest commit.

**Comment by Tetralux** [comment]
Fixed on both platforms as of d7a5767aa3497f9b4b6f0ced955ce31fb019ef80.
